protocols/rest-jakarta/src/main/java/org/jboss/arquillian/protocol/rest/RESTURIHandler.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 Red Hat Inc. and/or its affiliates and other contributors
+ * identified by the Git commit log. 
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,  
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.arquillian.protocol.rest;
+
+import java.lang.reflect.Method;
+import java.net.URI;
+import java.util.Collection;
+
+import org.jboss.arquillian.container.spi.client.protocol.metadata.HTTPContext;
+import org.jboss.arquillian.protocol.servlet5.ServletProtocolConfiguration;
+import org.jboss.arquillian.protocol.servlet5.ServletURIHandler;
+import org.jboss.arquillian.protocol.servlet5.ServletUtil;
+
+public class RESTURIHandler extends ServletURIHandler {
+
+    private ServletProtocolConfiguration config;
+    public RESTURIHandler(ServletProtocolConfiguration config, Collection<HTTPContext> contexts) {
+        super(config, contexts);
+        this.config = config;
+    }
+
+    public URI locateTestServlet(Method method) {
+        HTTPContext context = locateHTTPContext(method);
+        return ServletUtil.determineBaseURI(
+            config,
+            context,
+            RESTMethodExecutor.ARQUILLIAN_REST_NAME);
```

Comment:
The RESTProtocolApplication has an ApplicationPath of that ARQUILLIAN_REST_NAME similar to what is done for ARQUILLIAN_SERVLET_NAME.  A HttpContext has "servlet"'s defined in it, but it is a name for the servlet which can be anything as long as it can be found like what is done here and a context path for that "servlet".  If there is another http SPI that should be used instead for this to be configured, I will gladly move over to it, but I thought this was the best way to get this to work using the existing Arquillian SPIs.

As to wanting to note it, where would it be best to note it?

Reply:
Okay yeah. That's what I figured out. I'm trying to implement it right now in WildFly Arquillian. I think for simplicity it makes sense really. Either way a container needs to sort out the paths so I think what you have is good.

That's a good question on where to note it. Maybe once we have a few implementations it won't matter much.

Reply:
As you can see in the test cases for this protocol, I did use wildfly to test it out.  So hopefully that can help with running, but it was pretty straight forward to update the liberty plugin as you can see in my previously mentioned liberty arquillian PR.

Reply:
Yeah. It's pretty straight forward for WildFly too. I'm just about there I think :) Just deciding how much to hard-code it now. Thanks for this effort BTW.

Reply:
The whole reason for the adding of the servlet is to get the context root.  If that could be gotten another way, the URIHandler would not be needed.  We know what the REST protocol's application name is.  I will see if I can play with this tomorrow.

Reply:
Sure, that makes sense to me. I know in WildFly we read our management model to determine the context root which seems acceptable to me as it's likely container specific.

Reply:
OK so 2 votes to put it in.  Just need a committer to approve the review and merge it I think.  I know @starksm64 was looking at helping get this through the process.  After that it would be good to get this into a new 1.7.0 Alpha release.